### PR TITLE
Better API for Partial Fixed Base

### PR DESCRIPTION
**Pull request by EwanGilligan**
One of the desired features was being able to create a stabilizer chain from a partial base. This was possible, but it was a bit awkward using the base selector combinator. Since this is a common use case I thought it'd be good to add this feature, both a new selector (that is just a partial base selector with fixed base as the first base selector) and as a function to create stabilizer chains from.

I did think think of adding a partial base with strategy, but thought it may pollute the namespace a bit much - maybe let me know your thoughts. The new base selector should make things a bit more simple with the general stabchain with strategy or selector.

### CODE REVIEW

**Comment by EwanGilligan** [comment]
This pull request is dependent upon #86 , as otherwise clippy with complain a lot

File: examples/group_import.rs [hunk 1]
```diff
@@ -20,11 +20,10 @@ struct GAPGroup {
 
 impl GAPGroup {
     fn to_group(self) -> Group {
-        Group::from_iter(
-            self.generators
-                .into_iter()
-                .map(|images| ClassicalPermutation::from_slice(&images[..]).into()),
-        )
+        self.generators
```
Comment (WizardOfMenlo) [inline examples/group_import.rs:23]: With this we can remove the FromIter import at the top

File: src/group/brute_force.rs [hunk 2]
```diff
@@ -17,7 +16,11 @@ pub fn group_elements<P: Permutation>(g: &Group<P>) -> Vec<P> {
     let gens = g.generators();
     let mut res = DetHashSet::with_capacity_and_hasher(gens.len(), Default::default());
     // Get everything that is not the identity
-    let mut to_check = VecDeque::from_iter(gens.iter().cloned().filter(|p| !p.is_id()));
+    let mut to_check = gens
+        .iter()
+        .cloned()
+        .filter(|p| !p.is_id())
+        .collect::<VecDeque<P>>();
```
Comment (WizardOfMenlo) [inline src/group/brute_force.rs:23]: Personally, I prefer the `to_check: VecDeque<_>` style, but up to you

File: src/group/stabchain/builder/ift.rs [hunk 3]
```diff
@@ -98,7 +97,7 @@ where
         // Gets the record to be updated
         let mut record = self.chain[self.current_pos].clone();
 
-        let mut to_check = VecDeque::from_iter(record.transversal.keys().cloned());
+        let mut to_check = record.transversal.keys().cloned().collect::<VecDeque<_>>();
```
Comment (WizardOfMenlo) [inline src/group/stabchain/builder/ift.rs:100]: As above

File: src/group/stabchain/builder/ift.rs [hunk 4]
```diff
@@ -139,7 +138,7 @@ where
         }
 
         // We now want to check all the newly added elements
-        let mut to_check = VecDeque::from_iter(new_transversal.keys().cloned());
+        let mut to_check = new_transversal.keys().cloned().collect::<VecDeque<_>>();
```
Comment (WizardOfMenlo) [inline src/group/stabchain/builder/ift.rs:141]: As above

File: src/group/stabchain/builder/naive.rs [hunk 5]
```diff
@@ -99,7 +98,7 @@ where
         // Gets the record to be updated
         let mut record = self.chain[self.current_pos].clone();
 
-        let mut to_check = VecDeque::from_iter(record.transversal.keys().cloned());
+        let mut to_check = record.transversal.keys().cloned().collect::<VecDeque<_>>();
```
Comment (WizardOfMenlo) [inline src/group/stabchain/builder/naive.rs:101]: As above

File: src/group/stabchain/builder/naive.rs [hunk 6]
```diff
@@ -124,8 +123,10 @@ where
         }
 
         // We now want to check all the newly added elements
-        let mut to_check =
-            VecDeque::from_iter(new_transversal.iter().map(|(o, p)| (o.clone(), p.clone())));
+        let mut to_check = new_transversal
```
Comment (WizardOfMenlo) [inline src/group/stabchain/builder/naive.rs:126]: As above

File: src/group/stabchain/builder/random/random_ift.rs [hunk 7]
```diff
@@ -176,7 +176,7 @@ where
         let mut record = self.chain[self.current_pos].clone();
         //debug_assert!(record.gens.generators().contains(&p));
         // If this element is already a generator, then we can exit
-        let mut to_check = VecDeque::from_iter(record.transversal.keys().cloned());
+        let mut to_check = record.transversal.keys().cloned().collect::<VecDeque<_>>();
```
Comment (WizardOfMenlo) [inline src/group/stabchain/builder/random/random_ift.rs:179]: Ditto

File: src/group/stabchain/builder/random/random_ift.rs [hunk 8]
```diff
@@ -191,7 +191,7 @@ where
         }
 
         // We now want to check all the newly added elements
-        let mut to_check = VecDeque::from_iter(new_transversal.keys().cloned());
+        let mut to_check = new_transversal.keys().cloned().collect::<VecDeque<_>>();
```
Comment (WizardOfMenlo) [inline src/group/stabchain/builder/random/random_ift.rs:194]: Ditto

**Review by WizardOfMenlo — CHANGES_REQUESTED** [review]
Just a couple of stylistic changes, I made then on #86 but they did not carry over, nothing major I like the change overall

File: examples/group_import.rs (hunk 1 above)
Comment (EwanGilligan) [reply examples/group_import.rs]: I've merged the changes from #86 , so these should be resolved now

**Pull request merged**